Library/Services/Implementation/NfieldQuotaService.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Newtonsoft.Json;
+using Nfield.Extensions;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.SDK.Models;
+using Nfield.Services;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Nfield.SDK.Services.Implementation
+{
+    internal class NfieldQuotaService : INfieldQuotaService, INfieldConnectionClientObject
+    {
+        #region Implementation of INfieldConnectionClientObject
+
+        public INfieldConnectionClient ConnectionClient { get; internal set; }
+
+        public void InitializeNfieldConnection(INfieldConnectionClient connection)
+        {
+            ConnectionClient = connection;
+        }
+
+        #endregion
+
+        #region Implementation of INfieldQuotaService
+
+        /// <summary>
+        /// See <see cref="INfieldQuotaService.GetQuotaFrameVersionsAsync"/>
+        /// </summary>
+        public Task<IEnumerable<QuotaFrameVersion>> GetQuotaFrameVersionsAsync(string surveyId)
+        {
+            ValidateSurveyId(surveyId);
+
+            return Client.GetAsync(QuotaFrameVersionsUri(surveyId))
+                         .ContinueWith(task => task.Result.Content.ReadAsStringAsync().Result)
```

Comment:
I though we could do async await in the SDK already 🤔

Reply:
We can, probably this was copied from an older service. I will see if we can refactor it.